app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -73,6 +75,10 @@ class ProfileListAdapter(
             val current: Calendar = Calendar.getInstance()
             val currentHour = current.get(Calendar.HOUR_OF_DAY)
             val currentMinute = current.get(Calendar.MINUTE)
+            val currentDay = current.get(Calendar.DAY_OF_WEEK)
+            val type by lazy { object : TypeToken<List<Boolean>>() {}.type }
```

Comment:
Make a function in Utils.kt which takes a string and returns the MutableList<Boolean> to get the selected days.
Remove the repeated code from ProfileListAdapter and NewProfileFragment.

Reply:
@rob729 ok, I'll do so